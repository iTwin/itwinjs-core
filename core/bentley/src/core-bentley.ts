/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
export * from "./AccessToken";
export * from "./Assert";
export * from "./BeEvent";
export * from "./BentleyError";
export * from "./BentleyLoggerCategory";
export * from "./StatusCategory";
export * from "./BeSQLite";
export * from "./ByteStream";
export * from "./ClassUtils";
export * from "./Compare";
export * from "./CompressedId64Set";
export * from "./Dictionary";
export * from "./Disposable";
export * from "./Id";
export * from "./IndexMap";
export * from "./JsonSchema";
export * from "./JsonUtils";
export * from "./Logger";
export * from "./LRUMap";
export * from "./ObservableSet";
export * from "./OneAtATimeAction";
export * from "./OrderedId64Iterable";
export * from "./OrderedSet";
export * from "./partitionArray";
export * from "./PriorityQueue";
export * from "./ProcessDetector";
export * from "./SortedArray";
export * from "./StringUtils";
export * from "./Time";
export * from "./Tracing";
export * from "./TupleKeyedMap";
export * from "./TypedArrayBuilder";
export * from "./UnexpectedErrors";
export * from "./UtilityTypes";
export * from "./YieldManager";

// Temporarily (until 5.0) export top-level internal APIs to avoid breaking callers.
export * from "./internal/cross-package";

/** @docs-package-description
 * The core-bentley package contains classes to solve problems that are common for both client and server use cases.
 */
/**
 * @docs-group-description BeSQLite
 * Classes for working with SQLite databases. SQLite underlies IModelDb and ECDb - see [Executing ECSQL]($docs/learning/ECSQL.md)
 */
/**
 * @docs-group-description Errors
 * Classes for working with errors.
 */
/**
 * @docs-group-description Events
 * Classes for raising and handling events.
 */
/**
 * @docs-group-description Ids
 * Classes for working with unique identifiers.
 */
/**
 * @docs-group-description Logging
 * Classes for configuring and logging diagnostic messages - see [Learning about Logging]($docs/learning/common/Logging.md)
 */
/**
 * @docs-group-description Collections
 * Specialized, customizable collection classes like priority queues.
 */
/**
 * @docs-group-description Json
 * utilities for dealing with Json strings and files.
 */
/**
 * @docs-group-description Utils
 * Miscellaneous utility classes.
 */
/**
 * @docs-group-description ProcessDetector
 * Functions for determining the type of the current JavaScript process.
 */
