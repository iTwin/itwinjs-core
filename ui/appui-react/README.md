# @itwin/appui-react

Copyright Â© Bentley Systems, Incorporated. All rights reserved. See LICENSE.md for license terms and full copyright notice.

## Description

The __@itwin/appui-react__ package contains application fragments for Login, Project, iModel and View selection, and configuration of the application UI including the Backstage, Frontstages, Widgets, etc.

The AppUI React package is a portion of the iTwin.js User Interface library. It consists of major fragments of code that combine the
functionality from the `@itwin/core-frontend` with the core-react, imodel-components and appui-layout-react packages to implement substantial
functionality that can be used within an iTwin.js application.

The package makes use of React for rendering of the user interface, and uses react-redux and its concept of the state store for
actions using reducers.

There are a number of sub-frameworks, each which has a set of components, actions, and reducers, and contributes to the overall State. Each such
sub-framework is contained in a separate folder.

## Documentation

See the [iTwin.js](https://www.itwinjs.org) documentation for more information.
