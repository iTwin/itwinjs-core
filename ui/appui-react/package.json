{
  "name": "@itwin/appui-react",
  "version": "3.3.0-dev.31",
  "description": "UI framework",
  "main": "lib/cjs/appui-react.js",
  "module": "lib/esm/appui-react.js",
  "typings": "lib/cjs/appui-react",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/iTwin/itwinjs-core/tree/master/ui/appui-react"
  },
  "scripts": {
    "build": "npm run -s copy:locale && npm run -s build:cjs",
    "build:ci": "npm run -s build && npm run -s build:esm",
    "build:cjs": "npm run -s copy:cjs && tsc -p tsconfig.cjs.json",
    "build:esm": "npm run -s copy:esm && tsc -p tsconfig.esm.json",
    "build:watch": "run-p watch:cjs watch:esm",
    "watch:cjs": "npm run -s build:cjs -- -w",
    "watch:esm": "npm run -s build:esm -- -w",
    "copy:locale": "cpx \"./public/**/*\" ./lib/public",
    "copy:cjs": "cpx \"./src/**/*.{*css,json,svg}\" \"./lib/cjs\"",
    "copy:esm": "cpx \"./src/**/*.{*css,json,svg}\" \"./lib/esm\"",
    "pseudolocalize": "betools pseudolocalize --englishDir ./public/locales/en --out ./public/locales/en-PSEUDO",
    "clean": "rimraf lib .rush/temp/package-deps*.json",
    "cover": "nyc npm -s test",
    "docs": "betools docs --includes=../../generated-docs/extract --json=../../generated-docs/ui/appui-react/file.json --excludeGlob=**/declarations.d.ts --tsIndexFile=./appui-react.ts --onlyJson",
    "lint": "eslint -f visualstudio \"./src/**/*.{ts,tsx}\" 1>&2",
    "extract-api": "betools extract-api --entry=appui-react",
    "test": "mocha --config ../.mocharc.json \"./lib/cjs/test/**/*.test.js\"",
    "test:watch": "npm -s test -- --reporter min --watch-extensions ts,tsx --watch"
  },
  "keywords": [
    "Bentley",
    "BIM",
    "iModel"
  ],
  "author": {
    "name": "Bentley Systems, Inc.",
    "url": "http://www.bentley.com"
  },
  "peerDependencies": {
    "@itwin/appui-abstract": "workspace:^3.3.0-dev.31",
    "@itwin/appui-layout-react": "workspace:^3.3.0-dev.31",
    "@itwin/components-react": "workspace:^3.3.0-dev.31",
    "@itwin/core-bentley": "workspace:^3.3.0-dev.31",
    "@itwin/core-common": "workspace:^3.3.0-dev.31",
    "@itwin/core-frontend": "workspace:^3.3.0-dev.31",
    "@itwin/core-geometry": "workspace:^3.3.0-dev.31",
    "@itwin/core-markup": "workspace:^3.3.0-dev.31",
    "@itwin/core-quantity": "workspace:^3.3.0-dev.31",
    "@itwin/core-react": "workspace:^3.3.0-dev.31",
    "@itwin/imodel-components-react": "workspace:^3.3.0-dev.31",
    "@itwin/presentation-common": "workspace:^3.3.0-dev.31",
    "@itwin/presentation-frontend": "workspace:^3.3.0-dev.31",
    "react": "^17.0.0",
    "react-dom": "^17.0.0",
    "react-redux": "^7.2.2",
    "redux": "^4.1.0"
  },
  "//devDependencies": [
    "NOTE: All peerDependencies should also be listed as devDependencies since peerDependencies are not considered by npm install",
    "NOTE: All tools used by scripts in this package must be listed as devDependencies"
  ],
  "devDependencies": {
    "@itwin/appui-abstract": "workspace:*",
    "@itwin/appui-layout-react": "workspace:*",
    "@itwin/build-tools": "workspace:*",
    "@itwin/components-react": "workspace:*",
    "@itwin/core-bentley": "workspace:*",
    "@itwin/core-common": "workspace:*",
    "@itwin/core-frontend": "workspace:*",
    "@itwin/core-geometry": "workspace:*",
    "@itwin/core-markup": "workspace:*",
    "@itwin/core-quantity": "workspace:*",
    "@itwin/core-react": "workspace:*",
    "@itwin/eslint-plugin": "workspace:*",
    "@itwin/imodel-components-react": "workspace:*",
    "@itwin/presentation-common": "workspace:*",
    "@itwin/presentation-frontend": "workspace:*",
    "@itwin/presentation-testing": "workspace:*",
    "@testing-library/react": "^12.0.0",
    "@testing-library/react-hooks": "^7.0.2",
    "@types/chai": "^4.1.4",
    "@types/chai-as-promised": "^7",
    "@types/chai-jest-snapshot": "^1.3.0",
    "@types/chai-spies": "^1.0.0",
    "@types/enzyme": "3.9.3",
    "@types/faker": "^4.1.0",
    "@types/lodash": "^4.14.0",
    "@types/mocha": "^8.2.2",
    "@types/node": "16.11.7",
    "@types/react": "^17.0.37",
    "@types/react-dom": "^17.0.0",
    "@types/react-redux": "^7.1.18",
    "@types/rimraf": "^2.0.2",
    "@types/sinon": "^9.0.0",
    "@types/sinon-chai": "^3.2.0",
    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.3",
    "chai": "^4.1.2",
    "chai-as-promised": "^7",
    "chai-jest-snapshot": "^2.0.0",
    "chai-spies": "1.0.0",
    "cpx2": "^3.0.0",
    "enzyme": "3.10.0",
    "enzyme-to-json": "^3.3.4",
    "eslint": "^7.11.0",
    "faker": "^4.1.0",
    "ignore-styles": "^5.0.1",
    "jsdom": "^19.0.0",
    "jsdom-global": "3.0.2",
    "mocha": "^10.0.0",
    "npm-run-all": "^4.1.5",
    "nyc": "^15.1.0",
    "raf": "^3.4.0",
    "react": "^17.0.0",
    "react-dom": "^17.0.0",
    "react-redux": "^7.2.2",
    "react-test-renderer": "^17.0.0",
    "redux": "^4.1.0",
    "rimraf": "^3.0.2",
    "sinon": "^9.0.2",
    "sinon-chai": "^3.2.0",
    "ts-node": "^10.4.0",
    "typemoq": "^2.1.0",
    "typescript": "~4.4.0",
    "xmlhttprequest": "^1.8.0"
  },
  "//dependencies": [
    "NOTE: these dependencies should be only for things that DO NOT APPEAR IN THE API",
    "NOTE: core-frontend should remain UI technology agnostic, so no react/angular dependencies are allowed"
  ],
  "dependencies": {
    "@bentley/icons-generic": "^1.0.15",
    "@bentley/icons-generic-webfont": "^1.0.15",
    "@itwin/core-telemetry": "workspace:*",
    "@itwin/presentation-components": "workspace:*",
    "@itwin/itwinui-css": "0.x",
    "@itwin/itwinui-react": "^1.38.1",
    "@itwin/itwinui-icons-react": "^1.8.0",
    "classnames": "^2.3.1",
    "immer": "9.0.6",
    "lodash": "^4.17.10",
    "react-dnd": "^11.1.3",
    "react-dnd-html5-backend": "^11.1.3",
    "rxjs": "^6.6.2"
  },
  "nyc": {
    "extends": "./node_modules/@itwin/build-tools/.nycrc",
    "require": [
      "ignore-styles",
      "jsdom-global/register",
      "source-map-support/register",
      "ts-node/register"
    ],
    "check-coverage": true,
    "statements": 99,
    "branches": 99,
    "functions": 99,
    "lines": 99
  },
  "eslintConfig": {
    "plugins": [
      "@itwin"
    ],
    "extends": [
      "plugin:@itwin/ui",
      "plugin:@itwin/jsdoc"
    ]
  }
}