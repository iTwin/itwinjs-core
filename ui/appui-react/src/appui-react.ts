/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/

// cSpell:ignore safearea cursormenu clientservices oidc Textbox Modeless configurableui stagepanels dragdrop uiadmin itemsarbiter Popout

export * from "./appui-react/UiFramework";  // Please ensure that this line comes before all other exports.

export * from "./appui-react/accudraw/AccuDrawCommandItems";
export * from "./appui-react/accudraw/AccuDrawDialog";
export * from "./appui-react/accudraw/AccuDrawKeyboardShortcuts";
export * from "./appui-react/accudraw/AccuDrawPopupManager";
export * from "./appui-react/accudraw/AccuDrawUiSettings";
export * from "./appui-react/accudraw/AccuDrawWidget";
export * from "./appui-react/accudraw/Calculator";
export * from "./appui-react/accudraw/CalculatorEngine";
export * from "./appui-react/accudraw/CalculatorPopup";
export * from "./appui-react/accudraw/FrameworkAccuDraw";
export * from "./appui-react/accudraw/MenuButton";
export * from "./appui-react/accudraw/MenuButtonPopup";

export * from "./appui-react/backstage/Backstage";
export * from "./appui-react/backstage/BackstageComposer";
export * from "./appui-react/backstage/BackstageComposerItem";
export * from "./appui-react/backstage/BackstageItemProps";
export * from "./appui-react/backstage/BackstageItemUtilities";
export * from "./appui-react/backstage/BackstageManager";
export * from "./appui-react/backstage/CommandLaunch";
export * from "./appui-react/backstage/FrontstageLaunch";
export * from "./appui-react/backstage/Separator";
export * from "./appui-react/backstage/useDefaultBackstageItems";
export * from "./appui-react/backstage/TaskLaunch";
export * from "./appui-react/backstage/useUiItemsProviderBackstageItems";

export * from "./appui-react/configurableui/ConfigurableUiContent";
export * from "./appui-react/configurableui/ConfigurableUiControl";
export * from "./appui-react/configurableui/ConfigurableUiManager";
export * from "./appui-react/configurableui/state";

export * from "./appui-react/content/ContentControl";
export * from "./appui-react/content/ContentGroup";
export * from "./appui-react/content/ContentLayout";
export * from "./appui-react/content/ContentLayoutManager";
export * from "./appui-react/content/ContentViewManager";
export * from "./appui-react/content/ViewStateHelper";
export * from "./appui-react/content/StageContentLayout";
export * from "./appui-react/content/ViewportContentControl";
export * from "./appui-react/content/IModelViewport";
export * from "./appui-react/content/DefaultViewOverlay";

export * from "./appui-react/cursor/CursorInformation";
export * from "./appui-react/cursor/cursorprompt/CursorPrompt";
export * from "./appui-react/cursor/cursorpopup/CursorPopup";
export * from "./appui-react/cursor/cursorpopup/CursorPopupManager";
export * from "./appui-react/cursor/cursormenu/CursorMenu";

export * from "./appui-react/dialog/DialogManagerBase";
export * from "./appui-react/dialog/ModalDialogManager";
export * from "./appui-react/dialog/ModelessDialog";
export * from "./appui-react/dialog/ModelessDialogManager";
export * from "./appui-react/dialog/StandardMessageBox";
export * from "./appui-react/dialog/UiDataProvidedDialog";

export * from "./appui-react/feedback/ValidationTextbox";
export * from "./appui-react/feedback/ElementTooltip";

export * from "./appui-react/frontstage/Frontstage";
export * from "./appui-react/frontstage/FrontstageComposer";
export * from "./appui-react/frontstage/FrontstageDef";
export * from "./appui-react/frontstage/FrontstageManager";
export * from "./appui-react/frontstage/FrontstageProvider";
export * from "./appui-react/frontstage/NestedFrontstage";
export * from "./appui-react/frontstage/ModalFrontstage";
export * from "./appui-react/frontstage/ModalSettingsStage";
export * from "./appui-react/frontstage/StandardFrontstageProvider";

export * from "./appui-react/hooks/useActiveIModelConnection";
export * from "./appui-react/hooks/useActiveStageId";
export * from "./appui-react/hooks/useActiveViewport";
export * from "./appui-react/hooks/useAvailableUiItemsProviders";
export * from "./appui-react/hooks/useAnalysisAnimationDataProvider";
export * from "./appui-react/hooks/useFrameworkVersion";
export * from "./appui-react/hooks/useScheduleAnimationDataProvider";
export * from "./appui-react/hooks/useSolarDataProvider";

export * from "./appui-react/imodel-components/spatial-tree/SpatialContainmentTree";
export * from "./appui-react/imodel-components/category-tree/CategoriesTree";
export * from "./appui-react/imodel-components/category-tree/CategoryVisibilityHandler";
export * from "./appui-react/imodel-components/models-tree/ModelsTree";
export * from "./appui-react/imodel-components/models-tree/ModelsVisibilityHandler";
export * from "./appui-react/imodel-components/Common";
export * from "./appui-react/imodel-components/VisibilityTreeEventHandler";
export * from "./appui-react/imodel-components/VisibilityTreeRenderer";

export * from "./appui-react/keyboardshortcut/KeyboardShortcut";
export * from "./appui-react/keyboardshortcut/KeyboardShortcutMenu";
export * from "./appui-react/keyinbrowser/KeyinBrowser";

export * from "./appui-react/messages/ActivityMessage";
export * from "./appui-react/messages/ActivityMessagePopup";
export * from "./appui-react/messages/AppNotificationManager";
export * from "./appui-react/messages/InputField";
export * from "./appui-react/messages/MessageManager";
export * from "./appui-react/messages/StatusMessageRenderer";
export * from "./appui-react/messages/Pointer";
export * from "./appui-react/messages/ReactNotifyMessageDetails";
export * from "./appui-react/messages/StickyMessage";
export * from "./appui-react/messages/ToastMessage";

export * from "./appui-react/navigationaids/CubeNavigationAidControl";
export * from "./appui-react/navigationaids/DrawingNavigationAidControl";
export * from "./appui-react/navigationaids/NavigationAidControl";
export * from "./appui-react/navigationaids/SheetNavigationAid";
export * from "./appui-react/navigationaids/SheetsModalFrontstage";
export * from "./appui-react/navigationaids/StandardRotationNavigationAid";

export * from "./appui-react/pickers/ListPicker";
export * from "./appui-react/pickers/ViewSelector";

export * from "./appui-react/childwindow/ChildWindowManager";

export * from "./appui-react/popup/KeyinPalettePanel";
export * from "./appui-react/popup/KeyinPalettePopup";
export * from "./appui-react/popup/HTMLElementPopup";
export * from "./appui-react/popup/InputEditorPopup";
export * from "./appui-react/popup/PopupManager";
export * from "./appui-react/popup/PositionPopup";
export * from "./appui-react/popup/ToolbarPopup";

export * from "./appui-react/redux/SessionState";
export * from "./appui-react/redux/StateManager";
export * from "./appui-react/redux/FrameworkState";
export * from "./appui-react/redux/connectIModel";
export * from "./appui-react/redux/ReducerRegistry";
export * from "./appui-react/redux/redux-ts";

export * from "./appui-react/safearea/SafeAreaContext";

export * from "./appui-react/selection/SelectionContextItemDef";
export * from "./appui-react/selection/HideIsolateEmphasizeManager";
export * from "./appui-react/selection/ClearEmphasisStatusField";

export * from "./appui-react/settings/ui/UiSettingsPage";
export * from "./appui-react/settings/quantityformatting/QuantityFormat";
export * from "./appui-react/settings/quantityformatting/UnitSystemSelector";

export * from "./appui-react/shared/ActionButtonItemDef";
export * from "./appui-react/shared/AnyItemDef";
export * from "./appui-react/shared/AnyToolbarItemDef";
export * from "./appui-react/shared/CommandItemDef";
export * from "./appui-react/shared/CustomItemDef";
export * from "./appui-react/shared/CustomItemProps";
export * from "./appui-react/shared/GroupItemProps";
export * from "./appui-react/shared/ItemDefBase";
export * from "./appui-react/shared/ItemMap";
export * from "./appui-react/shared/ItemProps";
export * from "./appui-react/shared/MenuItem";
export * from "./appui-react/shared/SelectionScope";
export * from "./appui-react/shared/ToolItemDef";

export * from "./appui-react/stagepanels/FrameworkStagePanel";
export * from "./appui-react/stagepanels/StagePanel";
export * from "./appui-react/stagepanels/StagePanelDef";
export * from "./appui-react/stagepanels/StagePanelHeader";

export * from "./appui-react/statusbar/StatusBar";
export * from "./appui-react/statusbar/StatusBarWidgetControl";
export * from "./appui-react/statusbar/StatusBarComposer";
export * from "./appui-react/statusbar/StatusBarItem";
export * from "./appui-react/statusbar/StatusBarItemsManager";
export * from "./appui-react/statusbar/StatusBarItemUtilities";
export * from "./appui-react/statusbar/StatusBarComposer";
export * from "./appui-react/statusbar/withMessageCenterFieldProps";
export * from "./appui-react/statusbar/useUiItemsProviderStatusBarItems";
export * from "./appui-react/statusbar/useDefaultStatusBarItems";
export * from "./appui-react/statusbar/withStatusFieldProps";

export * from "./appui-react/statusfields/tileloading/TileLoadingIndicator";
export * from "./appui-react/statusfields/ActivityCenter";
export * from "./appui-react/statusfields/ConditionalField";
export * from "./appui-react/statusfields/FooterModeField";
export * from "./appui-react/statusfields/Indicator";
export * from "./appui-react/statusfields/MessageCenter";
export * from "./appui-react/statusfields/PromptField";
export * from "./appui-react/statusfields/SectionsField";
export * from "./appui-react/statusfields/SelectionInfo";
export * from "./appui-react/statusfields/SelectionScope";
export * from "./appui-react/statusfields/SnapMode";
export * from "./appui-react/statusfields/StatusFieldProps";
export * from "./appui-react/statusfields/ViewAttributes";
export * from "./appui-react/statusfields/toolassistance/ToolAssistanceField";

export * from "./appui-react/syncui/SyncUiEventDispatcher";
export * from "./appui-react/syncui/BooleanListener";

export * from "./appui-react/theme/ThemeManager";

export * from "./appui-react/timeline/ScheduleAnimationProvider";
export * from "./appui-react/timeline/AnalysisAnimationProvider";
export * from "./appui-react/timeline/SolarTimelineDataProvider";

export * from "./appui-react/toolbar/ActionButtonItem";
export * from "./appui-react/toolbar/ActionItemButton";
export * from "./appui-react/toolbar/DragInteraction";
export * from "./appui-react/toolbar/ToolbarComposer";
export * from "./appui-react/toolbar/GroupButtonItem";
export * from "./appui-react/toolbar/GroupItem";
export * from "./appui-react/toolbar/PopupButton";
export * from "./appui-react/toolbar/Toolbar";
export * from "./appui-react/toolbar/ToolbarHelper";
export * from "./appui-react/toolbar/ToolButton";
export * from "./appui-react/toolbar/useUiItemsProviderToolbarItems";
export * from "./appui-react/toolbar/useDefaultToolbarItems";

export * from "./appui-react/tools/CoreToolDefinitions";
export * from "./appui-react/tools/FrameworkToolAdmin";
export * from "./appui-react/tools/MarkupToolDefinitions";
export * from "./appui-react/tools/RestoreLayoutTool";
export * from "./appui-react/tools/ToolSettingsTools";

export * from "./appui-react/uiadmin/FrameworkUiAdmin";

export * from "./appui-react/ui-items-provider/StandardContentToolsProvider";
export * from "./appui-react/ui-items-provider/StandardNavigationToolsProvider";
export * from "./appui-react/ui-items-provider/StandardStatusbarItemsProvider";
export * from "./appui-react/uiprovider/DefaultDialogGridContainer";

export * from "./appui-react/uistate/AppUiSettings";
export * from "./appui-react/uistate/UserSettingsStorage";
export * from "./appui-react/uistate/useUiStateStorage";

export * from "./appui-react/utils/ViewUtilities";
export * from "./appui-react/utils/PropsHelper";
export * from "./appui-react/utils/UiShowHideManager";
export * from "./appui-react/utils/ToolbarButtonHelper";

export * from "./appui-react/widget-panels/Content";
export * from "./appui-react/widget-panels/Frontstage";
export * from "./appui-react/widget-panels/FrontstageContent";
export * from "./appui-react/widget-panels/ModalFrontstageComposer";
export * from "./appui-react/widget-panels/StatusBar";
export * from "./appui-react/widget-panels/Tab";
export * from "./appui-react/widget-panels/Toolbars";
export * from "./appui-react/widget-panels/ToolSettings";
export * from "./appui-react/widget-panels/useWidgetDirection";

export * from "./appui-react/widgets/BackstageAppButton";
export * from "./appui-react/widgets/BasicNavigationWidget";
export * from "./appui-react/widgets/BasicToolWidget";
export * from "./appui-react/widgets/DefaultNavigationWidget";
export * from "./appui-react/widgets/NavigationWidget";
export * from "./appui-react/widgets/NavigationWidgetComposer";
export * from "./appui-react/widgets/ReviewToolWidget";
export * from "./appui-react/widgets/ViewToolWidgetComposer";
export * from "./appui-react/widgets/StatusBarWidgetComposerControl";
export * from "./appui-react/widgets/ContentToolWidgetComposer";
export * from "./appui-react/widgets/StableWidgetDef";
export * from "./appui-react/widgets/ToolbarWidgetBase";
export * from "./appui-react/widgets/ToolWidget";
export * from "./appui-react/widgets/ToolWidgetComposer";
export * from "./appui-react/widgets/Widget";
export * from "./appui-react/widgets/WidgetControl";
export * from "./appui-react/widgets/WidgetDef";
export * from "./appui-react/widgets/WidgetHost";
export * from "./appui-react/widgets/WidgetManager";
export * from "./appui-react/widgets/WidgetProps";
export * from "./appui-react/widgets/WidgetStack";

export * from "./appui-react/workflow/Task";
export * from "./appui-react/workflow/Workflow";

export * from "./appui-react/zones/FrameworkZone";
export * from "./appui-react/zones/StatusBarZone";
export * from "./appui-react/zones/toolsettings/ToolSettingsZone";
export * from "./appui-react/zones/toolsettings/ToolInformation";
export * from "./appui-react/zones/toolsettings/ToolSettingsManager";
export * from "./appui-react/zones/toolsettings/ToolUiProvider";
export * from "./appui-react/zones/toolsettings/DefaultToolSettingsProvider";
export * from "./appui-react/zones/Zone";
export * from "./appui-react/zones/ZoneDef";

/** @docs-package-description
 * The ui-framework package contains classes and components for specifying the application UI consisting of the
 * Backstage, Frontstages, Content Views, Tool Bars, Status Bars, Widgets and Panels.
 * For more information, see [learning about ui-framework]($docs/learning/ui/framework/index.md).
 */
/**
 * @docs-group-description AccuDraw
 * Classes and components providing a UI for AccuDraw, an aide for entering coordinate data.
 */
/**
 * @docs-group-description Admin
 * APIs for various UI components, such as toolbars, buttons and menus.
 */
/**
 * @docs-group-description Backstage
 * Classes for working with a Backstage
 */
/**
 * @docs-group-description ConfigurableUi
 * Classes for working with the Application UI Configuration
 */
/**
 * @docs-group-description ContentView
 * Classes for working with a Content View, Group, Layout or Control
 */
/**
 * @docs-group-description Cursor
 * Cursor related information, components and events
 */
/**
 * @docs-group-description Dialog
 * Classes for working with a dialog
 */
/**
 * @docs-group-description Frontstage
 * Classes for working with a Frontstage
 */
/**
 * @docs-group-description Hooks
 * Hook functions for use in Functional React Components.
 */
/**
 * @docs-group-description IModelComponents
 * Classes for displaying information about an iModel
 */
/**
 * @docs-group-description Item
 * Classes for working with an Item in a Tool Widget, Navigation Widget or Backstage
 */
/**
 * @docs-group-description KeyboardShortcut
 * Classes for working with Keyboard Shortcuts
 */
/**
 * @docs-group-description NavigationAids
 * Classes for working with Navigation Aids
 */
/**
 * @docs-group-description Notification
 * Classes for working with a Notification or Message
 */
/**
 * @docs-group-description ChildWindowManager
 * Classes for working with child windows.
 */
/**
 * @docs-group-description Picker
 * Classes for working with various pickers
 */
/**
 * @docs-group-description Popup
 * Classes for working with popup components
 */
/**
 * @docs-group-description State
 * Classes for maintaining state
 */
/**
 * @docs-group-description Settings
 * Classes and components used by settings pages displayed in the modal settings stage
 */
/**
 * @docs-group-description StandardUiItemsProvider
 * Standard UiItemsProvider classes.
 */
/**
 * @docs-group-description StatusBar
 * Classes for defining a StatusBar
 */
/**
 * @docs-group-description SyncUi
 * Classes for informing UI components to sync/refresh their display
 */
/**
 * @docs-group-description Timeline
 * Classes for working with a TimelineComponent
 */
/**
 * @docs-group-description Toolbar
 * Classes used to construct a Toolbar
 */
/**
 * @docs-group-description Tools
 * Core Tool definitions
 */
/**
 * @docs-group-description ToolSettings
 * Classes for working Tool Settings.  See learning documentation [Tool Settings]($docs/learning/ui/framework/toolsettings.md).
 */
/**
 * @docs-group-description UiProvider
 * Interfaces and classes for specifying UI items to be inserted at runtime.
 */
/**
 * @docs-group-description UiStateStorage
 * Interfaces and classes for persisting UI settings.
 */
/**
 * @docs-group-description Utilities
 * Various utility classes for working with a UI and Redux.
 */
/**
 * @docs-group-description Widget
 * Classes for working with a Widget
 */
/**
 * @docs-group-description Zone
 * Classes for working with a Zone
 */
