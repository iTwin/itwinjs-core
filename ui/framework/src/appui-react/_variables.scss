/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/core-react/lib/core-react/style/themecolors";

$uifw-popup-border-color: $buic-background-panel-stroke;
$uifw-popup-border-width: 1px;
$uifw-popup-border-radius: 3px;
$uifw-popup-content-padding: 5px;
$uifw-popup-title-bar-min-height: 24px;
$uifw-popup-opacity: 0.80;