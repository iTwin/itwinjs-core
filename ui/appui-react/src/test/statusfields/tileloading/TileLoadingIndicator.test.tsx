/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
import { shallow } from "enzyme";
import * as React from "react";
import * as moq from "typemoq";
import { BeEvent } from "@itwin/core-bentley";
import { IModelApp, MockRender, ScreenViewport, Viewport } from "@itwin/core-frontend";
import { TileLoadingIndicator } from "../../../appui-react";
import TestUtils, { mount } from "../../TestUtils";
import { cleanup, render } from "@testing-library/react";
import { expect } from "chai";

describe("TileLoadingIndicator", () => {

  before(async () => {
    await TestUtils.initializeUiFramework();
    await MockRender.App.startup();
  });

  after(async () => {
    await MockRender.App.shutdown();
    TestUtils.terminateUiFramework();
  });

  it("should render correctly as footer by default", () => {
    const wrapper = render(<TileLoadingIndicator />);

    expect(wrapper.container.querySelector(".nz-footer-mode")).to.exist;
    cleanup();
  });

  it("should render correctly footer (deprecated)", () => {
    shallow(
      <TileLoadingIndicator isInFooterMode={true} />,
    ).should.matchSnapshot();
  });

  it("should render correctly not footer (deprecated)", () => {
    shallow(
      <TileLoadingIndicator isInFooterMode={false} />,
    ).should.matchSnapshot();
  });

  it("should unmount correctly", () => {
    const sut = mount(<TileLoadingIndicator />);
    sut.unmount();
  });

  it("50% then 100% complete", async () => {
    // numReadyTiles / (numReadyTiles + numRequestedTiles)
    const onRenderEvent = new BeEvent<(vp: Viewport) => void>();
    const viewportMock = moq.Mock.ofType<ScreenViewport>();

    // added because component registers interest in onRender events
    viewportMock.setup((x) => x.onRender).returns(() => onRenderEvent);

    await IModelApp.viewManager.setSelectedView(viewportMock.object);
    const wrapper = mount(<TileLoadingIndicator />);
    IModelApp.viewManager.onViewOpen.emit(viewportMock.object);
    // 10% complete
    viewportMock.setup((viewport) => viewport.numRequestedTiles).returns(() => 90);
    viewportMock.setup((viewport) => viewport.numReadyTiles).returns(() => 10);
    onRenderEvent.raiseEvent(viewportMock.object);
    await TestUtils.flushAsyncOperations();

    // 50% complete
    viewportMock.setup((viewport) => viewport.numRequestedTiles).returns(() => 250);
    viewportMock.setup((viewport) => viewport.numReadyTiles).returns(() => 250);
    onRenderEvent.raiseEvent(viewportMock.object);
    await TestUtils.flushAsyncOperations();

    // 100% complete
    viewportMock.setup((viewport) => viewport.numRequestedTiles).returns(() => 0);
    viewportMock.setup((viewport) => viewport.numReadyTiles).returns(() => 0);
    onRenderEvent.raiseEvent(viewportMock.object);
    await TestUtils.flushAsyncOperations();

    wrapper.update();
    wrapper.unmount();
  });
});
