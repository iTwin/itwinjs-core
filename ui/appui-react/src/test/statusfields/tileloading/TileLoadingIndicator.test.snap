// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileLoadingIndicator should render correctly footer 1`] = `
<div
  className="uifw-tile-loading-bar nz-footer-mode"
>
  <span />
  <ProgressLinear
    value={0}
  />
</div>
`;

exports[`TileLoadingIndicator should render correctly not footer 1`] = `
<div
  className="uifw-tile-loading-bar"
>
  <span />
  <ProgressLinear
    value={0}
  />
</div>
`;
